metricbeat/module/couchbase/cluster/data.go

```diff
@@ -0,0 +1,71 @@
+package cluster
+
+import (
+	"encoding/json"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"io"
+)
+
+type StorageTotals_Ram struct {
+	Total             int64 `json:"total"`
+	QuotaTotal        int64 `json:"quotaTotal"`
+	QuotaUsed         int64 `json:"quotaUsed"`
+	Used              int64 `json:"used"`
+	UsedByData        int64 `json:"usedByData"`
+	QuotaUsedPerNode  int64 `json:"quotaUsedPerNode"`
+	QuotaTotalPerNode int64 `json:"quotaTotalPerNode"`
+}
+type StorageTotals_Hdd struct {
+	Total      int64 `json:"total"`
+	QuotaTotal int64 `json:"quotaTotal"`
+	Used       int64 `json:"used"`
+	UsedByData int64 `json:"usedByData"`
+	Free       int64 `json:"free"`
+}
+
+type StorageTotals struct {
+	RAM StorageTotals_Ram `json:"ram"`
+	Hdd StorageTotals_Hdd `json:"hdd"`
+}
+
+type Data struct {
+	StorageTotals        StorageTotals `json:"storageTotals"`
+	IndexMemoryQuota     int64         `json:"indexMemoryQuota"`
+	MemoryQuota          int64         `json:"memoryQuota"`
+	RebalanceStatus      string        `json:"rebalanceStatus"`
+	RebalanceProgressURI string        `json:"rebalanceProgressUri"`
+	StopRebalanceURI     string        `json:"stopRebalanceUri"`
+	NodeStatusesURI      string        `json:"nodeStatusesUri"`
+	MaxBucketCount       int64         `json:"maxBucketCount"`
+}
+
+func eventMapping(body io.Reader) common.MapStr {
+
+	var d Data
+	err := json.NewDecoder(body).Decode(&d)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	logp.Info("Printing Data:")
+	event := common.MapStr{
+		"quota.index_memory.mb":          d.IndexMemoryQuota,
```

Comment:
The reason it is possible do use dots in the fields.yml is because we do some "dedotting" magic in our python script. This is mainly to make writing the fields.yml easier. We don't have that in Golang. So yes, each dotted entry should end up in a nested document for compatibility reason with elasticsearch 2.x.